### PR TITLE
New `gh_token_exists()`

### PR DESCRIPTION
**Pull request by hadley**
I think this will be useful for improving some error messages in usethis.

### CODE REVIEW

**Review by gaborcsardi — APPROVED** [review]
Thanks!

File: R/gh_token.R [hunk 1]
```diff
@@ -53,6 +53,20 @@ gh_token <- function(api_url = NULL) {
...
+gh_token_exists <- function(api_url = NULL) {
+  api_url <- api_url %||% default_api_url()
+  tryCatch(
+    {
+      token <- gitcreds::gitcreds_get(get_hosturl(api_url))
+      gh_pat(token$password)
+      TRUE
+    } ,
+    error = function(e) FALSE
+  )
```
Comment (jennybc) [inline R/gh_token.R:67]: Should this be factored in such a way that it can be used inside `gh_token()` just above?

**Review by jennybc — COMMENTED** [review]
> I think this will be useful for improving some error messages in usethis.

Just for context usethis does have a lot of existing logic re: detecting a PAT and identifying any problems with it (most important ones: exists but has expired, exists but is under-scoped).

https://github.com/r-lib/usethis/blob/2cc9e5a4e5e3034399fafb65d871eec352d83a15/R/github_token.R#L122-L220

File: R/gh_token.R (hunk 1 above)
Comment (hadley) [reply R/gh_token.R:67]: It's not immediately obvious to me how to do that due to a few subtle differences in the functions.

**Pull request merged**